Add announcement blog post and update homepage (#186)
symposium-dev/symposium: md/blog/announcing-symposium.md

# Announcing Symposium - AI The Rust Way

Are you using an AI agent to write Rust code (or curious to try it)? If so, GREAT NEWS! We'd like to share with you **Symposium** - a Rust-focused interoperability layer that connects AI agents to crate-authored skills, tools, and workflows.

(If you've read Niko's [previous][symposium-blog-1] [blog posts][symposium-blog-2] talking about Symposium, this is pretty different! The tool we're announcing today is the result of many iterations of figuring out what exactly the "thing we want" is. So please, read on!)

[symposium-blog-1]: https://smallcultfollowing.com/babysteps/blog/2025/09/24/symposium/
[symposium-blog-2]: https://smallcultfollowing.com/babysteps/blog/2025/10/08/symmacp/

## What is Symposium?

> Symposium is a tool that examines what crates your project depends on and uses that to automatically install new skills, MCP servers, or other extensions. These extensions help your AI agent to write better code, avoid common footguns and pitfalls, and even leverage tools like the [Rust Token Killer (RTK)](GREAT NEWS!) to save you tokens.

Reviewer: ```suggestion
Symposium is a tool that examines what crates your project depends on and uses that to automatically install new skills, MCP servers, or other extensions. These extensions help your AI agent to write better code, avoid common footguns and pitfalls, and leverage tools like the [Rust Token Killer (RTK)](https://rtk-ai.app) to save you tokens.
```

?

## Getting started

> The primary way to interact with Symposium today is through the `cargo agents` CLI command. If you want to try it, do this:

Reviewer: ```suggestion
You interact with Symposium through the `cargo agents` CLI command. If you want to try it, do this:
```

```rust
cargo binstall symposium # or `cargo install`
cargo agents init
```

The init command will prompt you to select what agents you want to use and a few other things. Based on that we install hooks that will cause Symposium to be invoked automatically. The next time you start an agent on a Rust project, Symposium will check if there are available skills or other extensions for the crates you use and set them up automatically. You shouldn't have to do anything else.

# Our goals and principles

You may be familiar with various extensions that agents can work with, such has [MCP servers](https://modelcontextprotocol.io/docs/getting-started/intro), [Skills](https://agentskills.io/home), or Hooks. You may also know that different agents have different levels of support for these, and even different takes on them (Hooks, for example, are not as well-standardized as MCP servers and Skills). However, that doesn't diminish the fact that many people have built many tools around these extension systems. We want you to *easily* use these ecosystem tools.

> You may *also* have run into cases where a model is "outdated" compared to either the Rust language itself (e.g. there may be a newer language feature that is more idiomatic) or was trained on an older versions of a crate that you are using. It's generally not hard to get models to follow newer conventions, but they need to be *told* to do so. We want to make that easier and more automated. 

Reviewer: ```suggestion
You may *also* have run into cases where a model is "outdated" compared to either the Rust language itself (e.g., there may be a newer language feature that is more idiomatic) or was trained on an older versions of a crate that you are using. It's generally not hard to get models to follow newer conventions, but they need to be *told* to do so. We want to make that easier and more automated. 
```

## MCP servers

MCP servers were one of the first extensions made available by agents. They expose a set of tools, either local or remote, that agents can call. MCP servers defined by a Symposium plugin get installed into your agent's settings for use.

# What's next?

Today, we're shipping a minimal, usual product for users to experiment with and hopefully find use from. But, we're far from done. We have a number of really interesting ideas to make Symposium even more useful.

We want to continue to expand the set of agent features that Symposium supports. When an agent supports a tool or similar, we want it to be a minimal process to be able to recommend that users of your crate also use that tool. Often, this means that we should "just install" those tools into project-local agent settings; but, we want to make sure that this is done *correctly* and supports the agents that our users use. However, we *also* want to support (when possible) more dynamic loading - such as by dispatching hooks through Symposium itself, or having Symposium register a transparent MCP server layer. There are lots of things we can do here, and we're excited to hear what people want and need first.

> We currently have pretty minimal support for how to run hooks or MCP servers - really just a command to run. We already have in-progress work to support declarative dependencies, which in turns allows both auto-installation and auto-updates. Using a symposium plugin should "just work".

Reviewer: I don't 100% know what you are referring to here -- I think this already works today, even? i.e., we'll autorefresh git repositories and the like.
